src/main.ts

```diff
@@ -1,11 +1,25 @@
 import log from "./log"
-import {getEnv} from "./env"
+import { getEnv } from "./env"
 import LastBlockStore from "./LastBlockStore"
-import {ethers} from "ethers"
-import {throwIfNotContract} from "./checkArguments"
+import { ethers } from "ethers"
+import { BigNumber } from "ethers/utils"
+import { throwIfNotContract } from "./checkArguments"
 import Watcher from "./watcher"
 import CoreAPIClient from "./CoreAPIClient"
+
 import MarketplaceJSON from "../lib/marketplace-contracts/build/contracts/Marketplace.json"
+import StreamRegistryJSON from "../lib/streamregistry/StreamRegistryV3.json"
+
+import { TransactionReceipt } from "ethers/providers"
+
+type EthereumAddress = string
+type Permission = {
```

Comment:
Why these are type aliases? Should't they be interfaces or classes to allow extensibility?